modules/aggregation-layers/src/aggregation-layer.js

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2015 - 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {CompositeLayer, AttributeManager, experimental} from '@deck.gl/core';
+import {cssToDeviceRatio} from '@luma.gl/core';
+const {compareProps} = experimental;
+
+// props when changed results in new uniforms that requires re-aggregation
+const UNIFORM_PROPS = [
+  // DATA-FILTER extension
+  'filterEnabled',
+  'filterRange',
+  'filterSoftRange',
+  'filterTransformSize',
+  'filterTransformColor'
+];
+
+export default class AggregationLayer extends CompositeLayer {
```

Comment:
Not using `Layer` as base class, doing so I have to copy `_renderLayers` method.